TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -389,6 +389,49 @@ public override bool Visit(DataDefinition dataDefinition)
 
             }
 
+            //Check if variable is Sync inside Strict Type definition
+            if (dataDefinition.SemanticData.HasFlag(Symbol.Flags.InsideTypedef) &&
+                dataDefinition.SemanticData.HasFlag(Symbol.Flags.Sync))
+            {
+                //Typedef instruction => check if it's marked Strict
+                if (dataDefinition.SemanticData.Kind == Symbol.Kinds.Typedef && dataDefinition.SemanticData.HasFlag(Symbol.Flags.Strict))
+                {
+                    DiagnosticUtils.AddError(dataDefinition, $"Cannot declare Type definition {dataDefinition.Name} with Sync clause because it is Strict.");
+                }
+                //Variable inside Typedef => check if parent typedef is marked Strict
+                else if (dataDefinition.SemanticData.NearestParent(Symbol.Kinds.Typedef).HasFlag(Symbol.Flags.Strict))
+                {
+                    DiagnosticUtils.AddError(dataDefinition, $"{dataDefinition.Name} cannot be declared with Sync clause because it is part of Strict Type definition {dataDefinition.ParentTypeDefinition?.Name}.");
+                }
+
+            }
+            
+            //Variable is of a user defined Strict Type
+            if (dataDefinition.SemanticData.HasFlag(Symbol.Flags.HasATypedefType) && dataDefinition.DataType.RestrictionLevel == RestrictionLevel.STRICT)
+            {
+                //Sync clause not allowed on variable declaration because its type is Strict
+                if (dataDefinition.SemanticData.HasFlag(Symbol.Flags.Sync))
+                {
+                    DiagnosticUtils.AddError(dataDefinition, $"{dataDefinition.Name} cannot be declared with Sync clause because its Type definition {dataDefinition.DataType.Name} is Strict.");
+                }
+                //Check whether variable is under a sync parent
+                else if (dataDefinition.CodeElement?.LevelNumber?.Value > 1)
```

Comment:
dataDefinition.SemanticData can be converted to VariableSymbol which has a Level property. Start by casting dataDefinition.SemanticData so you can use it all the way in the checker.